### PR TITLE
UT-3125 Relative path of texture fix

### PR DESCRIPTION
**Pull request by nat-w**
Temp fbx file is now created in the directory where the exported fbx file will be created so that the relative path to textures is correct.

### CODE REVIEW

**Comment by nat-w** [comment]
Previously:
before

Now:
after

File: com.unity.formats.fbx/Editor/FbxExporter.cs [hunk 1]
```diff
@@ -3438,17 +3438,23 @@ internal enum TransformExportType { Local, Global, Reset };
             Dictionary<GameObject, IExportData> exportData)
         {
             exportCancelled = false;
+            
+            m_lastFilePath = LastFilePath;
```
Comment (lassond) [inline com.unity.formats.fbx/Editor/FbxExporter.cs:3442]: Why did you move this statement?

File: com.unity.formats.fbx/Editor/FbxExporter.cs [hunk 2]
```diff
@@ -3438,17 +3438,23 @@ internal enum TransformExportType { Local, Global, Reset };
...
             // This way we won't overwrite existing files.
-            try{
-                m_tempFilePath = Path.GetTempFileName();
+            try
+            {
+                // create a temp file in the same directory where the fbx will be exported
+                var exportDir = Path.GetDirectoryName(m_lastFilePath);
+                var tempFileName = Path.GetFileNameWithoutExtension(Path.GetRandomFileName());
+                // have to add .fbx to the end or else the temp file path won't match the file name
+                m_tempFilePath = exportDir + "/" + tempFileName + ".fbx";
```
Comment (lassond) [inline com.unity.formats.fbx/Editor/FbxExporter.cs:3453]: Here you should use `Path.Combine`. This will play better with other platforms

File: com.unity.formats.fbx/Editor/FbxExporter.cs (hunk 1 above)
Comment (lassond) [reply com.unity.formats.fbx/Editor/FbxExporter.cs:3442]: oh, I understand now, you use it below

File: com.unity.formats.fbx/Editor/FbxExporter.cs (hunk 2 above)
Comment (lassond) [reply com.unity.formats.fbx/Editor/FbxExporter.cs:3453]: I would also add the actual file name in m_tempFilePath. The metadata will at least give a hint about the right file name.

For example:
```
fileName = Path.GetFileName(m_lastFilePath);
m_tempFilePath = Path.Combine(exportDir, tempFileName + "_" + fileName);
```
